app/models/scheme.rb

```diff
@@ -211,11 +209,16 @@ def validate_confirmed
     end
   end
 
+  def required_attributes
+    attribute_names - %w[id created_at updated_at old_id old_visible_id confirmed end_date sensitive secondary_client_group total_units has_other_client_group deactivation_date deactivation_date_type]
+  end
+
   def available_from
     created_at
   end
 
   def status
+    return :incomplete if required_attributes.any? { |attribute| self[attribute].blank? }
```

Comment:
```suggestion
    return :incomplete unless confirmed
```

Is this not the same thing?

Reply:
Ah, nice I hadn't realise that was what confirmed was actually testing